components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -246,6 +247,61 @@ sealed class TrackingProtectionAction : BrowserAction() {
     data class ClearTrackersAction(val tabId: String) : TrackingProtectionAction()
 }
 
+/**
+ * [BrowserAction] implementations related to updating BrowserState.extensions] and TabState.extensions.
+ */
+sealed class WebExtensionAction : BrowserAction() {
+
+    /**
+     * [BrowserAction] implementations related to updating the
+     * [mozilla.components.browser.state.state.BrowserState.extensions]
+     */
+    sealed class BrowserState : WebExtensionAction() {
```

Comment:
We could flatten this into just three actions, but it depends on how you want to implement the reducer. If we keep the action groups / nested groups , let's *not* call them "*State" though. That's pretty confusing to read, as the reducer applies actions to state :)